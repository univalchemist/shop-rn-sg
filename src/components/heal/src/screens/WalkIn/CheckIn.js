import React, { useEffect, useState } from 'react';
import { connect } from 'react-redux';
import {
  SafeAreaView,
  Dimensions,
  StyleSheet,
  TouchableOpacity,
  Image,
} from 'react-native';
import moment from 'moment';
import PropTypes from 'prop-types';

import { useIntl, useTheme } from '@heal/src/wrappers/core/hooks';
import { Box, PlainText } from '@heal/src/wrappers/components';
import { DOCTOR_LANDING } from '@routes';

import theme from '@theme';
import { location, iconDoctor } from '@heal/images';

const screenWidth = Dimensions.get('window').width;

const styles = StyleSheet.create({
  title: {
    marginTop: 24,
    fontStyle: 'normal',
    fontSize: 32,
    lineHeight: 37,
    letterSpacing: -1.5,
    fontWeight: '300',
    textAlign: 'center',
    color: theme.colors.gray[0],
  },
  contentBox: {
    marginHorizontal: 32,
    marginTop: 34,
    width: screenWidth - 32 * 2,
    backgroundColor: theme.colors.white,
    shadowOpacity: 0.05,
    shadowOffset: { width: 0, height: 2 },
    shadowRadius: 6,
    shadowColor: theme.colors.black,
  },
  text: {
    fontSize: 14,
    lineHeight: 20,
    letterSpacing: 0.25,
    color: theme.colors.gray[2],
  },
  backButton: {
    width: screenWidth - 32 * 2,
    height: 48,
    alignItems: 'center',
    justifyContent: 'center',
    borderWidth: 1,
    borderColor: theme.colors.gray[0],
    marginHorizontal: 32,
    position: 'absolute',
    bottom: 88,
  },
});

const CheckIn = ({ route, navigation, qrCodeData, checkedInData }) => {
  const theme = useTheme();
  const intl = useIntl();
  const { selectedDoctor } = route.params;

  const parseDateTimeToTime = time => {
    return moment(time, 'YYYY-MM-DDThh:mm:ss').format('hh:mm A');
  };

  const parseDateTimeToDate = time => {
    return moment(time, 'YYYY-MM-DDThh:mm:ss').format('DD MMM YYYY');
  };

  return (
    <Box flex={1} as={SafeAreaView}>
      <PlainText fontSize={32} style={styles.title}>
        {intl.formatMessage({ id: 'heal.walkInCheckIn.title' })}
      </PlainText>
      <Box style={styles.contentBox}>
        <Box mx={24} my={44} alignItems="center" justifyContent="center">
          <PlainText color={theme.colors.black} fontWeight="bold">
            {intl.formatMessage({
              id: 'heal.walkInCheckIn.estimatedConsultation',
            })}
          </PlainText>
          <PlainText style={styles.title}>
            {parseDateTimeToTime(checkedInData.arrivalTime)}
          </PlainText>
          <PlainText color={theme.colors.gray[2]} mt={8}>
            {parseDateTimeToDate(checkedInData.arrivalTime)}
          </PlainText>
          <Box flexDirection="row" mt={24} mr="auto" alignItems="center">
            <Image source={iconDoctor} />
            <Box ml={18}>
              <PlainText style={styles.text}>{selectedDoctor.name}</PlainText>
              <PlainText style={styles.text}>
                {selectedDoctor.specialityCode
                  ? intl.formatMessage({
                      id: `heal.speciality.name.${selectedDoctor.specialityCode}`,
                    })
                  : ''}
              </PlainText>
            </Box>
          </Box>
          <Box flexDirection="row" mt={24} mr="auto" alignItems="center">
            <Image source={location} />
            <PlainText ml={18} mr={18} style={styles.text} numberOfLines={4}>
              {qrCodeData.clinic.address}
            </PlainText>
          </Box>
        </Box>
      </Box>
      <TouchableOpacity
        style={styles.backButton}
        onPress={() => {
          navigation.navigate(DOCTOR_LANDING);
        }}
      >
        <PlainText color={theme.colors.gray[0]}>
          {intl.formatMessage({ id: 'heal.walkInCheckIn.back' })}
        </PlainText>
      </TouchableOpacity>
    </Box>
  );
};

const mapStateToProps = ({
  heal: {
    checkedInData,
    scannedData: { qrCodeData },
  },
}) => ({
  qrCodeData,
  checkedInData,
});

export default connect(mapStateToProps, {})(CheckIn);
