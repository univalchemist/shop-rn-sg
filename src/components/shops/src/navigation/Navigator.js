/* istanbul ignore file */
//TODO update unit test
import React from 'react';
import {
  createStackNavigator,
  TransitionPresets,
} from '@react-navigation/stack';
import { connect } from 'react-redux';
import { useIntl } from '@shops/wrappers/core/hooks';
import {
  HomeScreen,
  SelectCategoryScreen,
  FilterByScreen,
  SortByScreen,
  ProductDetailsScreen,
  ReviewScreen,
} from '@shops/screens/home';
import {
  AddAddressScreen,
  DeliveryAddressesScreen,
  EditAddressScreen,
} from '@shops/screens/deliveryAddress';
import { ModalBackButton } from '@shops/wrappers/components';
import { ShopHeader, ClearAllButton } from '@shops/components';
import {
  CartScreen,
  EditProductScreen,
  CheckoutScreen,
  PaymentScreen,
  PaymentSuccessScreen,
  PaymentFailedScreen,
  PaymentPendingScreen,
} from '@shops/screens/cart';
import { updateSortType, updateFilterTypes } from '@shops/store/actions';
import { StackBackButton } from '@shops/wrappers/components';
import * as routes from './routes';
import {
  OrderHistoryDetailScreen,
  OrderHistoryScreen,
  ReviewSubmittedScreen,
  TrackOrderScreen,
  WriteReviewScreen,
} from '@shops/screens/orderhistory';
import { SingleSelectModal } from '@wrappers/components/form';
import LogScreen from '@shops/screens/log/LogScreen';
import SelectAddressModal from '@shops/screens/home/ProductDetailsScreen/SelectAddressModal';

const Stack = createStackNavigator();

const Navigator = ({ updateSortType, updateFilterTypes }) => {
  const intl = useIntl();

  const onClearAllSortByPress = () => {
    updateSortType(null);
  };

  const onClearAllFilterByPress = () => {
    updateFilterTypes({});
  };

  return (
    <Stack.Navigator
      initialRouteName={routes.NAVIGATION_HOME_SCREEN_PATH}
      screenOptions={{ headerBackTitleVisible: false }}
      headerMode="float"
    >
      <Stack.Screen
        name={routes.NAVIGATION_HOME_SCREEN_PATH}
        component={HomeScreen}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
        })}
      />
      <Stack.Screen
        name={routes.NAVIGATION_HOME_PRODUCT_PATH}
        component={ProductDetailsScreen}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
        })}
      />
      <Stack.Screen
        name={routes.SHOP_CART_SCREEN}
        component={CartScreen}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
        })}
      />
      <Stack.Screen
        name={routes.SHOP_CHECKOUT_SCREEN}
        component={CheckoutScreen}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
        })}
      />
      <Stack.Screen
        name={routes.SHOP_EDIT_PRODUCT_SCREEN}
        component={EditProductScreen}
        options={({ route, navigation }) => {
          const onUpdateFinish = route?.params?.onUpdateFinish;
          return {
            title: intl.formatMessage({ id: 'shop.screenTitle.editProduct' }),
            headerLeft: () => (
              <StackBackButton
                icName={'close'}
                onPress={() => {
                  navigation.goBack();
                  onUpdateFinish?.();
                }}
              />
            ),
            ...TransitionPresets.ModalTransition,
          };
        }}
      />
      <Stack.Screen
        name={routes.SHOP_SELECT_CATEGORY_MODAL}
        component={SelectCategoryScreen}
        options={{
          title: intl.formatMessage({ id: 'shop.screenTitle.selectCategory' }),
          headerBackImage: () => <ModalBackButton />,
          ...TransitionPresets.ModalTransition,
        }}
      />
      <Stack.Screen
        name={routes.SHOP_SORT_MODAL}
        component={SortByScreen}
        options={({ navigation }) => ({
          title: intl.formatMessage({ id: 'shop.screenTitle.sortBy' }),
          headerBackImage: () => <ModalBackButton />,
          headerRight: () => (
            <ClearAllButton
              navigation={navigation}
              text={intl.formatMessage({ id: 'shop.common.clearAll' })}
              onPress={onClearAllSortByPress}
            />
          ),
          ...TransitionPresets.ModalTransition,
        })}
      />
      <Stack.Screen
        name={routes.SHOP_FILTER_MODAL}
        component={FilterByScreen}
        options={({ navigation }) => ({
          title: intl.formatMessage({ id: 'shop.screenTitle.filterBy' }),
          headerBackImage: () => <ModalBackButton />,
          headerRight: () => (
            <ClearAllButton
              navigation={navigation}
              text={intl.formatMessage({ id: 'shop.common.clearAll' })}
              onPress={onClearAllFilterByPress}
            />
          ),
          ...TransitionPresets.ModalTransition,
        })}
      />

      <Stack.Screen
        name={routes.SELECT_ADDRESS_MODAL}
        component={SelectAddressModal}
        options={() => ({
          title: intl.formatMessage({ id: 'shop.product.selectionLocation' }),
          headerBackImage: () => <ModalBackButton />,
          ...TransitionPresets.ModalTransition,
        })}
      />
      <Stack.Screen
        name={routes.SHOP_REVIEW_SCREEN}
        component={ReviewScreen}
        options={({ route }) => {
          const ratingCount = route.params?.reviews?.length || 0;
          const title = intl.formatMessage(
            {
              id: 'shop.review.title',
              defaultMessage: `Review(${ratingCount})`,
            },
            { ratingCount },
          );

          return {
            title,
            headerBackImage: () => <ModalBackButton />,
            ...TransitionPresets.ModalTransition,
          };
        }}
      />
      <Stack.Screen
        name={routes.SHOP_DELIVERY_ADDRESSES_SCREEN}
        component={DeliveryAddressesScreen}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
        })}
      />
      <Stack.Screen
        name={routes.SHOP_ADD_ADDRESS_SCREEN}
        component={AddAddressScreen}
        options={{
          title: intl.formatMessage({ id: 'shop.screenTitle.addNewAddress' }),
          headerBackImage: () => <ModalBackButton />,
          ...TransitionPresets.ModalTransition,
        }}
      />
      <Stack.Screen
        name={routes.SHOP_EDIT_ADDRESS_SCREEN}
        component={EditAddressScreen}
        options={{
          title: intl.formatMessage({ id: 'shop.screenTitle.editAddress' }),
          headerBackImage: () => <ModalBackButton />,
          ...TransitionPresets.ModalTransition,
        }}
      />
      <Stack.Screen
        name={routes.ORDER_HISTORY_SCREEN}
        component={OrderHistoryScreen}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
        })}
      />
      <Stack.Screen
        component={OrderHistoryDetailScreen}
        name={routes.ORDER_HISTORY_DETAIL_SCREEN}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
        })}
      />
      <Stack.Screen
        component={TrackOrderScreen}
        name={routes.TRACK_ORDER_SCREEN}
        options={{
          title: intl.formatMessage({
            id: 'shop.orderHistory.trackOrderTitle',
          }),
          headerBackImage: () => <ModalBackButton />,
          ...TransitionPresets.ModalTransition,
        }}
      />
      <Stack.Screen
        component={WriteReviewScreen}
        name={routes.WRITE_REVIEW_SCREEN}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
        })}
      />
      <Stack.Screen
        component={ReviewSubmittedScreen}
        name={routes.REVIEW_SUBMITTED_SCREEN}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
          gestureEnabled: false,
        })}
      />
      <Stack.Screen
        component={PaymentScreen}
        name={routes.PAYMENT_SCREEN}
        options={({ navigation }) => ({
          headerLeft: () => (
            <StackBackButton
              onPress={() => {
                navigation.navigate(routes.SHOP_CART_SCREEN);
              }}
            />
          ),
          title: intl.formatMessage({
            id: 'shop.payment.title',
          }),
        })}
      />
      <Stack.Screen
        component={PaymentSuccessScreen}
        name={routes.PAYMENT_SUCCESS_SCREEN}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
          gestureEnabled: false,
        })}
      />
      <Stack.Screen
        component={PaymentFailedScreen}
        name={routes.PAYMENT_FAILED_SCREEN}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
          gestureEnabled: false,
        })}
      />
      <Stack.Screen
        component={PaymentPendingScreen}
        name={routes.PAYMENT_PENDING_SCREEN}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
          gestureEnabled: false,
        })}
      />
      <Stack.Screen
        component={LogScreen}
        name={'LogScreen'}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
          gestureEnabled: false,
        })}
      />
      <Stack.Screen
        name={routes.SHOP_SINGLE_SELECT_MODAL}
        component={SingleSelectModal}
        options={({ route }) => ({
          title: route.params?.title,
          headerBackImage: ModalBackButton,
          ...TransitionPresets.ModalTransition,
        })}
      />
    </Stack.Navigator>
  );
};

export const ShopNavigator = connect(null, {
  updateSortType,
  updateFilterTypes,
})(Navigator);
