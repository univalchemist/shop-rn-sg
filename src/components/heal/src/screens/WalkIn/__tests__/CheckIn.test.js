import React from 'react';
import { Linking, Platform, TouchableOpacity } from 'react-native';
import { render, renderForTest } from '@testUtils';
import navigation from '@testUtils/__mocks__/navigation';
import { flushMicrotasksQueue, fireEvent } from 'react-native-testing-library';

import CheckIn from '../CheckIn';
import { DOCTOR_LANDING } from '@routes';

const qrCodeData = {
  clinic: {
    id: 506,
    name: 'Clinic Icity',
    latitude: 3.1267439,
    longitude: 101.6424801,
    qrCode: '2502',
    district: 'Klang',
    area: null,
  },
  doctors: [
    {
      isOpen: false,
      isRemoteFull: false,
      isRemoteTicketEnabled: true,
      isAppointmentEnabled: false,
      position: 9999,
      estimatedConsultationTime: '7:30 pm',
      ticketDepositAmount: 0.0,
      canGetRefundOnCancellation: false,
      cancellationGraceMinutes: 0,
      id: 472,
      name: 'Clinic Icity',
      specialityCode: 'generalpractitio',
      gender: 'Male',
      locale: 'en-HK',
    },
    {
      isOpen: false,
      isRemoteFull: false,
      isRemoteTicketEnabled: true,
      isAppointmentEnabled: true,
      position: 9999,
      estimatedConsultationTime: '7:30 pm',
      ticketDepositAmount: 0.0,
      canGetRefundOnCancellation: false,
      cancellationGraceMinutes: 0,
      id: 463,
      name: 'Yeo An Shin',
      specialityCode: 'generalpractitio',
      gender: 'Male',
      locale: 'en-HK',
    },
  ],
};

const checkedInData = {
  token: '304|1590425263|601',
  type: 'WalkIn',
  estimatedConsultationTime: 'TBC',
  state: 'STARTED',
  arrivalTime: '2020-05-27T00:47:43',
};

const selectedDoctor = {
  isOpen: false,
  isRemoteFull: false,
  isRemoteTicketEnabled: true,
  isAppointmentEnabled: true,
  position: 9999,
  estimatedConsultationTime: '7:30 pm',
  ticketDepositAmount: 0.0,
  canGetRefundOnCancellation: false,
  cancellationGraceMinutes: 0,
  id: 463,
  name: 'Yeo An Shin',
  specialityCode: 'generalpractitio',
  gender: 'Male',
  locale: 'en-HK',
};

describe('CheckIn', () => {
  test('Render CheckIn', async () => {
    const component = renderForTest(
      <CheckIn
        navigation={navigation}
        route={{ params: { selectedDoctor } }}
      />,
      {
        initialState: {
          heal: {
            scannedData: {
              qrCodeData,
            },
            checkedInData,
          },
        },
      },
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should press doctor card', async () => {
    const [screen] = render(
      <CheckIn
        navigation={navigation}
        route={{ params: { selectedDoctor } }}
      />,
      {
        initialState: {
          heal: {
            scannedData: {
              qrCodeData,
            },
            checkedInData,
          },
        },
      },
    );

    await flushMicrotasksQueue();

    const items = screen.queryAllByType(TouchableOpacity);
    fireEvent.press(items[0]);
    expect(navigation.navigate).toBeCalledWith(DOCTOR_LANDING);
  });
});
