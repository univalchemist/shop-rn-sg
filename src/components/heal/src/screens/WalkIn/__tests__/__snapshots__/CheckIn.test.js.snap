// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckIn Render CheckIn 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              enabled={false}
              hitSlop={
                Object {
                  "right": -700,
                }
              }
              maxDeltaY={20}
              minOffsetX={5}
              onHandlerStateChange={[Function]}
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <RCTSafeAreaView
                        emulateUnlessSupported={true}
                        flex={1}
                        style={
                          Array [
                            Object {
                              "flexBasis": 0,
                              "flexGrow": 1,
                              "flexShrink": 1,
                            },
                          ]
                        }
                      >
                        <Text
                          color="gray.1"
                          fontFamily="default"
                          fontSize={32}
                          letterSpacing={0.3}
                          lineHeight={22}
                          style={
                            Array [
                              Object {
                                "color": "#646464",
                                "fontFamily": "Arial",
                                "fontSize": 32,
                                "letterSpacing": 0.3,
                                "lineHeight": 22,
                              },
                              Object {
                                "color": "#212121",
                                "fontSize": 32,
                                "fontStyle": "normal",
                                "fontWeight": "300",
                                "letterSpacing": -1.5,
                                "lineHeight": 37,
                                "marginTop": 24,
                                "textAlign": "center",
                              },
                            ]
                          }
                        >
                          You're checked-in
                        </Text>
                        <View
                          style={
                            Array [
                              Object {},
                              Object {
                                "backgroundColor": "#ffffff",
                                "marginHorizontal": 32,
                                "marginTop": 34,
                                "shadowColor": "#000000",
                                "shadowOffset": Object {
                                  "height": 2,
                                  "width": 0,
                                },
                                "shadowOpacity": 0.05,
                                "shadowRadius": 6,
                                "width": 686,
                              },
                            ]
                          }
                        >
                          <View
                            alignItems="center"
                            justifyContent="center"
                            mx={24}
                            my={44}
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "justifyContent": "center",
                                  "marginBottom": 44,
                                  "marginLeft": 24,
                                  "marginRight": 24,
                                  "marginTop": 44,
                                },
                              ]
                            }
                          >
                            <Text
                              color="#000000"
                              fontFamily="default"
                              fontSize={16}
                              fontWeight="bold"
                              letterSpacing={0.3}
                              lineHeight={22}
                              style={
                                Array [
                                  Object {
                                    "color": "#000000",
                                    "fontFamily": "Arial",
                                    "fontSize": 16,
                                    "fontWeight": "700",
                                    "letterSpacing": 0.3,
                                    "lineHeight": 22,
                                  },
                                ]
                              }
                            >
                              Estimated consulation time
                            </Text>
                            <Text
                              color="gray.1"
                              fontFamily="default"
                              fontSize={16}
                              letterSpacing={0.3}
                              lineHeight={22}
                              style={
                                Array [
                                  Object {
                                    "color": "#646464",
                                    "fontFamily": "Arial",
                                    "fontSize": 16,
                                    "letterSpacing": 0.3,
                                    "lineHeight": 22,
                                  },
                                  Object {
                                    "color": "#212121",
                                    "fontSize": 32,
                                    "fontStyle": "normal",
                                    "fontWeight": "300",
                                    "letterSpacing": -1.5,
                                    "lineHeight": 37,
                                    "marginTop": 24,
                                    "textAlign": "center",
                                  },
                                ]
                              }
                            >
                              12:47 AM
                            </Text>
                            <Text
                              color="#727272"
                              fontFamily="default"
                              fontSize={16}
                              letterSpacing={0.3}
                              lineHeight={22}
                              mt={8}
                              style={
                                Array [
                                  Object {
                                    "color": "#727272",
                                    "fontFamily": "Arial",
                                    "fontSize": 16,
                                    "letterSpacing": 0.3,
                                    "lineHeight": 22,
                                    "marginTop": 8,
                                  },
                                ]
                              }
                            >
                              27 May 2020
                            </Text>
                            <View
                              alignItems="center"
                              flexDirection="row"
                              mr="auto"
                              mt={24}
                              style={
                                Array [
                                  Object {
                                    "alignItems": "center",
                                    "flexDirection": "row",
                                    "marginRight": "auto",
                                    "marginTop": 24,
                                  },
                                ]
                              }
                            >
                              <Image
                                source={
                                  Object {
                                    "testUri": "../../../src/components/heal/images/icon-doctor.png",
                                  }
                                }
                              />
                              <View
                                ml={18}
                                style={
                                  Array [
                                    Object {
                                      "marginLeft": 18,
                                    },
                                  ]
                                }
                              >
                                <Text
                                  color="gray.1"
                                  fontFamily="default"
                                  fontSize={16}
                                  letterSpacing={0.3}
                                  lineHeight={22}
                                  style={
                                    Array [
                                      Object {
                                        "color": "#646464",
                                        "fontFamily": "Arial",
                                        "fontSize": 16,
                                        "letterSpacing": 0.3,
                                        "lineHeight": 22,
                                      },
                                      Object {
                                        "color": "#727272",
                                        "fontSize": 14,
                                        "letterSpacing": 0.25,
                                        "lineHeight": 20,
                                      },
                                    ]
                                  }
                                >
                                  Yeo An Shin
                                </Text>
                                <Text
                                  color="gray.1"
                                  fontFamily="default"
                                  fontSize={16}
                                  letterSpacing={0.3}
                                  lineHeight={22}
                                  style={
                                    Array [
                                      Object {
                                        "color": "#646464",
                                        "fontFamily": "Arial",
                                        "fontSize": 16,
                                        "letterSpacing": 0.3,
                                        "lineHeight": 22,
                                      },
                                      Object {
                                        "color": "#727272",
                                        "fontSize": 14,
                                        "letterSpacing": 0.25,
                                        "lineHeight": 20,
                                      },
                                    ]
                                  }
                                >
                                  General Practitioner
                                </Text>
                              </View>
                            </View>
                            <View
                              alignItems="center"
                              flexDirection="row"
                              mr="auto"
                              mt={24}
                              style={
                                Array [
                                  Object {
                                    "alignItems": "center",
                                    "flexDirection": "row",
                                    "marginRight": "auto",
                                    "marginTop": 24,
                                  },
                                ]
                              }
                            >
                              <Image
                                source={
                                  Object {
                                    "testUri": "../../../src/components/heal/images/location.png",
                                  }
                                }
                              />
                              <Text
                                color="gray.1"
                                fontFamily="default"
                                fontSize={16}
                                letterSpacing={0.3}
                                lineHeight={22}
                                ml={18}
                                mr={18}
                                numberOfLines={4}
                                style={
                                  Array [
                                    Object {
                                      "color": "#646464",
                                      "fontFamily": "Arial",
                                      "fontSize": 16,
                                      "letterSpacing": 0.3,
                                      "lineHeight": 22,
                                      "marginLeft": 18,
                                      "marginRight": 18,
                                    },
                                    Object {
                                      "color": "#727272",
                                      "fontSize": 14,
                                      "letterSpacing": 0.25,
                                      "lineHeight": 20,
                                    },
                                  ]
                                }
                              />
                            </View>
                          </View>
                        </View>
                        <View
                          accessible={true}
                          focusable={true}
                          isTVSelectable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "alignItems": "center",
                              "borderColor": "#212121",
                              "borderWidth": 1,
                              "bottom": 88,
                              "height": 48,
                              "justifyContent": "center",
                              "marginHorizontal": 32,
                              "opacity": 1,
                              "position": "absolute",
                              "width": 686,
                            }
                          }
                        >
                          <Text
                            color="#212121"
                            fontFamily="default"
                            fontSize={16}
                            letterSpacing={0.3}
                            lineHeight={22}
                            style={
                              Array [
                                Object {
                                  "color": "#212121",
                                  "fontFamily": "Arial",
                                  "fontSize": 16,
                                  "letterSpacing": 0.3,
                                  "lineHeight": 22,
                                },
                              ]
                            }
                          >
                            Back to homepage
                          </Text>
                        </View>
                      </RCTSafeAreaView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
